pkgs/development/libraries/ncurses/default.nix

```diff
@@ -92,57 +102,58 @@ stdenv.mkDerivation rec {
+        rm $out/lib/libncurses$suffix.$dylibtype
+        echo "INPUT(libncurses$suffix.${abiVersion-extension} -ltinfo$suffix)" > $out/lib/libncurses$suffix.$dylibtype
+      fi
+    done
+
     for newsuffix in $suffixes ""; do
       # Create a non-abi versioned config util links
       ln -svf $cfg $dev/bin/ncurses$newsuffix-config
-
       # Allow for end users who #include <ncurses?w/*.h>
       ln -svf . $dev/include/ncurses$newsuffix
-
       for library in $libs; do
-        for dylibtype in so dll dylib; do
+        for dylibtype in so; do
+          if [ -e "$out/lib/lib$library$suffix.$dylibtype" ]; then
+            echo "INPUT(-l$library$suffix)" > $out/lib/lib$library$newsuffix.$dylibtype
+          fi
+        done
+        for dylibtype in dll dylib; do
           if [ -e "$out/lib/lib''${library}$suffix.$dylibtype" ]; then
             ln -svf lib''${library}$suffix.$dylibtype $out/lib/lib$library$newsuffix.$dylibtype
             ln -svf lib''${library}$suffix.${abiVersion-extension} $out/lib/lib$library$newsuffix.${abiVersion-extension}
-            if [ "ncurses" = "$library" ]
-            then
-              # make libtinfo symlinks
-              ln -svf lib''${library}$suffix.$dylibtype $out/lib/libtinfo$newsuffix.$dylibtype
-              ln -svf lib''${library}$suffix.${abiVersion-extension} $out/lib/libtinfo$newsuffix.${abiVersion-extension}
-            fi
           fi
         done
         for statictype in a dll.a la; do
           if [ -e "$out/lib/lib''${library}$suffix.$statictype" ]; then
             ln -svf lib''${library}$suffix.$statictype $out/lib/lib$library$newsuffix.$statictype
-            if [ "ncurses" = "$library" ]
-            then
-              # make libtinfo symlinks
-              ln -svf lib''${library}$suffix.$statictype $out/lib/libtinfo$newsuffix.$statictype
-            fi
           fi
         done
         ln -svf ''${library}$suffix.pc $dev/lib/pkgconfig/$library$newsuffix.pc
       done
     done
 
-    # move some utilities to $bin
-    # these programs are used at runtime and don't really belong in $dev
-    moveToOutput "bin/clear" "$out"
-    moveToOutput "bin/reset" "$out"
-    moveToOutput "bin/tabs" "$out"
-    moveToOutput "bin/tic" "$out"
-    moveToOutput "bin/tput" "$out"
-    moveToOutput "bin/tset" "$out"
-    moveToOutput "bin/captoinfo" "$out"
-    moveToOutput "bin/infotocap" "$out"
-    moveToOutput "bin/infocmp" "$out"
+    # Move some utilities to '$bin'.
+    #
+    # These programs are used at runtime and don't really belong in '$dev'.
+    moveToOutput 'bin/clear'     "$out"
+    moveToOutput 'bin/reset'     "$out"
+    moveToOutput 'bin/tabs'      "$out"
+    moveToOutput 'bin/tic'       "$out"
+    moveToOutput 'bin/tput'      "$out"
+    moveToOutput 'bin/tset'      "$out"
+    moveToOutput 'bin/captoinfo' "$out"
+    moveToOutput 'bin/infotocap' "$out"
+    moveToOutput 'bin/infocmp'   "$out"
```

Comment:
I'm getting lots of errors here that all look like:

```
Moving /nix/store/nrw7dh0d9pggn1ngf6l1v5rciagqjpr7-ncurses-6.2-dev/bin/clear to /nix/store/x8d5mqkmnsw5n27fpw5f0jba3dxixdln-ncurses-6.2/bin/clear
rmdir: failed to remove '/nix/store/nrw7dh0d9pggn1ngf6l1v5rciagqjpr7-ncurses-6.2-dev/bin': Directory not empty` errors
```

Is there a way to resolve this?